Launcher crashes on Android 14

### Describe the bug
The Middleware launcher crashes on Android 14 when the service is started, as it's missing the required `USE_EXACT_ALARM' permission.

### To Reproduce
- Run the App on Android 14
- Start the Middleware service via an intent

### Exceptions or error messages (if any)
```
06-14 10:07:32.824 14575 14575 I MonoDroid: UNHANDLED EXCEPTION:
--
06-14 10:07:32.857 14575 14575 I MonoDroid: Java.Lang.RuntimeException: Unable to start receiver eu.fiskaltrust.androidlauncher.http.Start: java.lang.SecurityException: Caller eu.fiskaltrust.androidlauncher.http.deutschebahn needs to hold android.permission.SCHEDULE_EXACT_ALARM or android.permission.USE_EXACT_ALARM to set exact alarms. ---> Java.Lang.SecurityException: Caller eu.fiskaltrust.androidlauncher.http.deutschebahn needs to hold android.permission.SCHEDULE_EXACT_ALARM or android.permission.USE_EXACT_ALARM to set exact alarms. ---> Android.OS.RemoteException: Remote stack trace:
06-14 10:07:32.857 14575 14575 I MonoDroid:   at com.android.server.alarm.AlarmManagerService$5.set(AlarmManagerService.java:3010)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.app.IAlarmManager$Stub.onTransact(IAlarmManager.java:181)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.os.Binder.execTransactInternal(Binder.java:1339)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.os.Binder.execTransact(Binder.java:1275)
06-14 10:07:32.857 14575 14575 I MonoDroid:
06-14 10:07:32.857 14575 14575 I MonoDroid:    --- End of inner exception stack trace ---
06-14 10:07:32.857 14575 14575 I MonoDroid:   at Java.Interop.JniEnvironment+InstanceMethods.CallVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x0006e] in :0
06-14 10:07:32.857 14575 14575 I MonoDroid:   at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x00036] in :0
06-14 10:07:32.857 14575 14575 I MonoDroid:   at Android.App.AlarmManager.SetExact (Android.App.AlarmType type, System.Int64 triggerAtMillis, Android.App.PendingIntent operation) [0x0005a] in <7340629091c14b6eb465e053dd50e2d0>:0
06-14 10:07:32.857 14575 14575 I MonoDroid:   at fiskaltrust.AndroidLauncher.Http.Broadcasting.StartLauncherBroadcastReceiver.OnReceive (Android.Content.Context context, Android.Content.Intent intent) [0x00156] in :0
06-14 10:07:32.857 14575 14575 I MonoDroid:   at Android.Content.BroadcastReceiver.n_OnReceive_Landroid_content_Context_Landroid_content_Intent_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_context, System.IntPtr native_intent) [0x00017] in <7340629091c14b6eb465e053dd50e2d0>:0
06-14 10:07:32.857 14575 14575 I MonoDroid:   at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPLL_V (_JniMarshal_PPLL_V callback, System.IntPtr jnienv, System.IntPtr klazz, System.IntPtr p0, System.IntPtr p1) [0x00005] in <7340629091c14b6eb465e053dd50e2d0>:0
06-14 10:07:32.857 14575 14575 I MonoDroid:    --- End of inner exception stack trace ---
06-14 10:07:32.857 14575 14575 I MonoDroid:   --- End of managed Java.Lang.RuntimeException stack trace ---
06-14 10:07:32.857 14575 14575 I MonoDroid: java.lang.RuntimeException: Unable to start receiver eu.fiskaltrust.androidlauncher.http.Start: java.lang.SecurityException: Caller eu.fiskaltrust.androidlauncher.http.deutschebahn needs to hold android.permission.SCHEDULE_EXACT_ALARM or android.permission.USE_EXACT_ALARM to set exact alarms.
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.app.ActivityThread.handleReceiver(ActivityThread.java:4459)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.app.ActivityThread.-$$Nest$mhandleReceiver(Unknown Source:0)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2257)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.os.Handler.dispatchMessage(Handler.java:106)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.os.Looper.loopOnce(Looper.java:205)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.os.Looper.loop(Looper.java:294)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.app.ActivityThread.main(ActivityThread.java:8229)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at java.lang.reflect.Method.invoke(Native Method)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:977)
06-14 10:07:32.857 14575 14575 I MonoDroid: Caused by: java.lang.SecurityException: Caller eu.fiskaltrust.androidlauncher.http.deutschebahn needs to hold android.permission.SCHEDULE_EXACT_ALARM or android.permission.USE_EXACT_ALARM to set exact alarms.
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.os.Parcel.createException(Parcel.java:3041)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.os.Parcel.readException(Parcel.java:3024)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.os.Parcel.readException(Parcel.java:2966)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.app.IAlarmManager$Stub$Proxy.set(IAlarmManager.java:311)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.app.AlarmManager.setImpl(AlarmManager.java:1084)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.app.AlarmManager.setImpl(AlarmManager.java:1044)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.app.AlarmManager.setExact(AlarmManager.java:808)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at eu.fiskaltrust.androidlauncher.http.Start.n_onReceive(Native Method)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at eu.fiskaltrust.androidlauncher.http.Start.onReceive(Start.java:30)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.app.ActivityThread.handleReceiver(ActivityThread.java:4450)
06-14 10:07:32.857 14575 14575 I MonoDroid:   ... 9 more
06-14 10:07:32.857 14575 14575 I MonoDroid: Caused by: android.os.RemoteException: Remote stack trace:
06-14 10:07:32.857 14575 14575 I MonoDroid:   at com.android.server.alarm.AlarmManagerService$5.set(AlarmManagerService.java:3010)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.app.IAlarmManager$Stub.onTransact(IAlarmManager.java:181)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.os.Binder.execTransactInternal(Binder.java:1339)
06-14 10:07:32.857 14575 14575 I MonoDroid:   at android.os.Binder.execTransact(Binder.java:1275)
06-14 10:07:32.857 14575 14575 I MonoDroid:
06-14 10:07:32.857 14575 14575 I MonoDroid:
```

--- a/src/fiskaltrust.AndroidLauncher.Grpc/Broadcasting/StartLauncherBroadcastReceiver.cs
+++ b/src/fiskaltrust.AndroidLauncher.Grpc/Broadcasting/StartLauncherBroadcastReceiver.cs
@@ -46,7 +46,15 @@ namespace fiskaltrust.AndroidLauncher.Grpc.Broadcasting
             var pending = PendingIntent.GetBroadcast(Application.Context, 0, alarmIntent, PendingIntentFlags.UpdateCurrent | PendingIntentFlags.Immutable);
 
             var alarmManager = (AlarmManager)Application.Context.GetSystemService(Context.AlarmService);
-            alarmManager.SetExact(AlarmType.ElapsedRealtime, SystemClock.ElapsedRealtime() + 100, pending);
+
+            if (Build.VERSION.SdkInt <= BuildVersionCodes.SV2)
+            {
+                alarmManager.SetExact(AlarmType.ElapsedRealtime, SystemClock.ElapsedRealtime() + 100, pending);
+            }
+            else
+            {
+                alarmManager.Set(AlarmType.ElapsedRealtime, SystemClock.ElapsedRealtime() + 100, pending);
+            }
         }
     }
 

--- a/src/fiskaltrust.AndroidLauncher.Grpc/Properties/AndroidManifest.xml
+++ b/src/fiskaltrust.AndroidLauncher.Grpc/Properties/AndroidManifest.xml
@@ -1,12 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="1"
-	android:versionName="1.0" package="eu.fiskaltrust.androidlauncher.grpc"
-	android:installLocation="auto">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="1" android:versionName="1.0" package="eu.fiskaltrust.androidlauncher.grpc" android:installLocation="auto">
 	<uses-sdk android:minSdkVersion="24" android:targetSdkVersion="33" />
-	<application android:allowBackup="true"
-		android:manageSpaceActivity="eu.fiskaltrust.androidlauncher.common.ManageSpaceActivity"
-		android:icon="@mipmap/ic_launcher" android:label="@string/app_name" android:supportsRtl="true"
-		android:theme="@style/AppTheme" android:extractNativeLibs="true"></application>
+	<application android:allowBackup="true" android:manageSpaceActivity="eu.fiskaltrust.androidlauncher.common.ManageSpaceActivity" android:icon="@mipmap/ic_launcher" android:label="@string/app_name" android:supportsRtl="true" android:theme="@style/AppTheme" android:extractNativeLibs="true"></application>
 	<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
 	<uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
 	<uses-permission android:name="android.permission.INTERNET" />
@@ -14,5 +9,5 @@
 	<uses-permission android:name="android.permission.ACCESS_MOCK_LOCATION" />
 	<uses-permission android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS" />
 	<uses-permission android:name="android.permission.SCHEDULE_EXACT_ALARM" android:maxSdkVersion="32" />
-	<uses-permission android:name="android.permission.USE_EXACT_ALARM" />
+	<uses-permission android:name="com.android.alarm.permission.SET_ALARM" />
 </manifest>

--- a/src/fiskaltrust.AndroidLauncher.Http/Broadcasting/StartLauncherBroadcastReceiver.cs
+++ b/src/fiskaltrust.AndroidLauncher.Http/Broadcasting/StartLauncherBroadcastReceiver.cs
@@ -46,7 +46,15 @@ namespace fiskaltrust.AndroidLauncher.Http.Broadcasting
             var pending = PendingIntent.GetBroadcast(Application.Context, 0, alarmIntent, PendingIntentFlags.UpdateCurrent | PendingIntentFlags.Immutable);
 
             var alarmManager = (AlarmManager)Application.Context.GetSystemService(Context.AlarmService);
-            alarmManager.SetExact(AlarmType.ElapsedRealtime, SystemClock.ElapsedRealtime() + 100, pending);
+
+            if (Build.VERSION.SdkInt <= BuildVersionCodes.SV2)
+            {
+                alarmManager.SetExact(AlarmType.ElapsedRealtime, SystemClock.ElapsedRealtime() + 100, pending);
+            }
+            else
+            {
+                alarmManager.Set(AlarmType.ElapsedRealtime, SystemClock.ElapsedRealtime() + 100, pending);
+            }
         }
     }
 

--- a/src/fiskaltrust.AndroidLauncher.Http/Properties/AndroidManifest.xml
+++ b/src/fiskaltrust.AndroidLauncher.Http/Properties/AndroidManifest.xml
@@ -1,18 +1,12 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="1"
-	android:versionName="1.0" package="eu.fiskaltrust.androidlauncher.http"
-	android:installLocation="auto">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="1" android:versionName="1.0" package="eu.fiskaltrust.androidlauncher.http" android:installLocation="auto">
 	<uses-sdk android:minSdkVersion="24" android:targetSdkVersion="33" />
-	<application android:allowBackup="true"
-		android:manageSpaceActivity="eu.fiskaltrust.androidlauncher.common.ManageSpaceActivity"
-		android:icon="@mipmap/ic_launcher" android:label="@string/app_name" android:supportsRtl="true"
-		android:theme="@style/AppTheme" android:extractNativeLibs="true"
-		android:usesCleartextTraffic="true"></application>
+	<application android:allowBackup="true" android:manageSpaceActivity="eu.fiskaltrust.androidlauncher.common.ManageSpaceActivity" android:icon="@mipmap/ic_launcher" android:label="@string/app_name" android:supportsRtl="true" android:theme="@style/AppTheme" android:extractNativeLibs="true" android:usesCleartextTraffic="true"></application>
 	<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
 	<uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
 	<uses-permission android:name="android.permission.INTERNET" />
 	<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
 	<uses-permission android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS" />
 	<uses-permission android:name="android.permission.SCHEDULE_EXACT_ALARM" android:maxSdkVersion="32" />
-	<uses-permission android:name="android.permission.USE_EXACT_ALARM" />
+	<uses-permission android:name="com.android.alarm.permission.SET_ALARM" />
 </manifest>

